src/Symfony/Component/HttpFoundation/HeaderUtils.php

```diff
@@ -167,7 +167,7 @@ public static function makeDisposition(string $disposition, string $filename, st
         }
 
         if ('' === $filenameFallback) {
-            $filenameFallback = $filename;
+            $filenameFallback = self::getFilenameFallback($filename);
```

Comment:
Tell me if I'm wrong: you proposal is basically this change - the rest is factoring the logic, isn't it?

Reply:
I thought this was not the intended behavior; I can create a fallback in local code, sure.